### ISSUE TITLE
cannot find virtualenv packages

### ISSUE THREAD

**Issue by dcsanx** [microsoft/pyright#702]
**Describe the bug**
I guess this has been reported before but venv modules still don't seem to be found

**To Reproduce**
create project with `venv` dir
add a `pyrightconfig.json`

```
{
  "include": [
    "src"
  ],
  
  "exclude": [
    "**/__pycache__"
  ],

  "executionEnvironments": [
    {
      "root": "src",
      "extraPaths": [
        "venv"
      ]
    }
  ],

  "typingsPath": "src/typestubs",
  "pythonVersion": "3.6",
  "pythonPlatform": "Darwin",
  "verboseOutput": true

}
```

**Expected behavior**
typechecking functions

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/58190733/83314795-06a5fe00-a1d1-11ea-8bf5-96f83f8e3318.png)

**VS Code extension or command-line**
VScode 1.1.39

**Additional context**
issues
https://github.com/microsoft/pyright/issues/52
https://github.com/microsoft/pyright/issues/30

sample config
https://github.com/microsoft/pyright/blob/master/docs/configuration.md

**Comment by dcsanx** [comment]
```sh
▶ tree . -L 3
.
├── Makefile
├── pyrightconfig.json
├── requirements.txt
├── src
│   ├── receive.py
│   └── send.py
└── venv
    ├── bin
    │   ├── activate
    │   ├── activate.csh
    │   ├── activate.fish
    │   ├── activate.ps1
    │   ├── activate.xsh
    │   ├── activate_this.py
    │   ├── easy_install
    │   ├── easy_install-3.7
    │   ├── pip
    │   ├── pip3
    │   ├── pip3.7
    │   ├── python -> python3.7
    │   ├── python-config
    │   ├── python3 -> python3.7
    │   ├── python3.7
    │   └── wheel
    ├── include
    │   └── python3.7m -> /usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/include/python3.7m
    └── lib
        └── python3.7

7 directories, 21 files
(venv)
dev/syntax/ramq  master ✗                                                                                                                      3d0h ✖ ⚑ ◒
▶ python --version
Python 3.7.7
(venv)
dev/syntax/ramq  master ✗                                                                                                                      3d0h ✖ ⚑ ◒
▶ which python
/Users/dcollier/dev/syntax/ramq/venv/bin/python
(venv)
dev/syntax/ramq  master ✗                                                                                                                      3d0h ✖ ⚑ ◒
▶
```

**Comment by erictraut** [comment]
The "extraPath" config parameter is for specifying additional search paths. You are using it to point to a Python environment, not a search path. Pyright won't be able to resolve imports because there are no Python modules in the "venv" directory.

To specify the Python environment that is in the venv directory, you need to set the "python.pythonPath" VS Code setting. This is the same setting that the VS Code Python extension uses, so if you have that extension installed, you can use the convenient environment picker that it provides.

For additional details about how pyright resolves imports, check out [this documentation](https://github.com/microsoft/pyright/blob/master/docs/import-resolution.md).

**Comment by dcsanx** [comment]
so all that stuff about virtualenvs should be removed? or it's an expected conflict?

> venvPath [path, optional]: Path to a directory containing one or more subdirectories, each of which contains a virtual environment. Each execution environment (see below for details) can refer to a different virtual environment. When used in conjunction with a venv setting (see below), pyright will search for imports in the virtual environment’s site-packages directory rather than the paths specified in PYTHONPATH.

> venv [string, optional]: Used in conjunction with the venvPath, specifies the virtual environment to use. Individual execution environments may override this setting.

**Comment by erictraut** [comment]
That feature is to handle the case where a team of developers has a collection of venvs on their respective computers and agree on a name for the venv that's used for a particular project. It allows each developer to specify their own directory that contains venvs but specify which venv is used on a project-by-project basis. If you're not working in a large team of developers, this configuration feature is probably of little or no use to you.

**Comment by delfick** [comment]
is it possible to make it use the current virtualenv? so I can just be in the correct virtualenv when I open up vim and it proceeds to start pyright?

**Comment by erictraut** [comment]
If you do not configure a venv, Pyright will attempt to discover the currently-configured Python environment by executing a "python3" shell command (or just "python" on Windows). So if you have a Python environment configured in your shell environment before you start the language server, it should do the right thing.

**Comment by delfick** [comment]
Actually, it does too. Sorry, I see what I thought it didn't. Thanks, sorry for the noise!
